### PR TITLE
chore(BaseInput): reusable, internal Input component

### PR DESCRIPTION
**Pull request by YossiSaadi**
https://monday.monday.com/boards/3532714909/pulses/6437205266

### CODE REVIEW

File: packages/core/src/components/BaseInput/BaseInput.tsx [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+      </div>
+    );
+  }
+);
+
+export default BaseInput;
+
+const App = () => {
+  return <BaseInput inputSize={"small"}></BaseInput>;
+};
```
Comment (talkor) [inline packages/core/src/components/BaseInput/BaseInput.tsx:35]: What's this?

Comment (YossiSaadi) [reply packages/core/src/components/BaseInput/BaseInput.tsx:35]: oops :)

File: packages/core/src/components/BaseInput/BaseInput.tsx
Comment (talkor) [inline packages/core/src/components/BaseInput/BaseInput.tsx:1]: I'd call it just Input

File: packages/core/src/components/BaseInput/BaseInput.types.ts
Comment (talkor) [inline packages/core/src/components/BaseInput/BaseInput.types.ts:1]: Yes yes yes!!

File: packages/core/src/components/BaseInput/BaseInput.module.scss [hunk 2]
```diff
@@ -0,0 +1,118 @@
+@import "../../styles/Typography";
+
+.wrapper {
+  width: 100%;
+  position: relative;
+  display: flex;
+  align-items: center;
+  gap: var(--spacing-small);
+  padding-block: var(--spacing-xs);
+  padding-inline: var(--spacing-medium) var(--spacing-xs);
```
Comment (talkor) [inline packages/core/src/components/BaseInput/BaseInput.module.scss:10]: Let's make it a standard?

File: packages/core/src/components/BaseInput/BaseInput.module.scss [hunk 3]
```diff
@@ -0,0 +1,118 @@
...
+    height: 48px;
+    padding-block: var(--spacing-small);
+  }
+
+  &.underline {
+    border-left: none;
+    border-right: none;
+    border-top: none;
+    padding: var(--spacing-small) var(--spacing-large) var(--spacing-small) var(--spacing-xs);
+    border-radius: 0 !important;
```
Comment (talkor) [inline packages/core/src/components/BaseInput/BaseInput.module.scss:41]: Necesarry?

Comment (YossiSaadi) [reply packages/core/src/components/BaseInput/BaseInput.module.scss]: The PR was still in draft state, copied some of the logic from `TextField`, removed unnecessary stuff now

File: packages/core/src/components/BaseInput/BaseInput.module.scss (hunk 2 above)
Comment (YossiSaadi) [reply packages/core/src/components/BaseInput/BaseInput.module.scss:10]: wdym?

File: packages/core/src/components/BaseInput/BaseInput.tsx
Comment (YossiSaadi) [reply packages/core/src/components/BaseInput/BaseInput.tsx:1]: Hmm let's discuss it
I'm leaving it open for having an Input component which would be exported from Vibe

**Comment by YossiSaadi** [comment]
added ability to hide stories here #2062 
make sure to merge this only after #2062 is merged

File: packages/core/src/components/BaseInput/BaseInput.module.scss (hunk 2 above)
Comment (talkor) [reply packages/core/src/components/BaseInput/BaseInput.module.scss:10]: Logical properties

Comment (YossiSaadi) [reply packages/core/src/components/BaseInput/BaseInput.module.scss:10]: hmm I agree
Chrome 87+ support
we don't have users with less than 90, but anyway when we upgrade browsers it should be supported

File: packages/core/src/components/BaseInput/BaseInput.module.scss [hunk 4]
```diff
@@ -0,0 +1,112 @@
...
+    &:hover {
+      border-color: var(--positive-color);
+    }
+
+    &:has(.input:active, .input:focus) {
+      border-color: var(--positive-color);
+    }
+  }
+
+  &:has(.input[aria-invalid="true"]) {
```
Comment (talkor) [inline packages/core/src/components/BaseInput/BaseInput.module.scss:79]: Didn't we say we can't use `has` for now?

File: packages/core/src/components/BaseInput/BaseInput.tsx [hunk 5]
```diff
@@ -0,0 +1,45 @@
...
+  (
+    {
+      size = "medium",
+      leftRender,
+      rightRender,
+      withReadOnlyStyle,
+      success,
+      error,
+      wrapperRole,
+      inputRole,
```
Comment (talkor) [inline packages/core/src/components/BaseInput/BaseInput.tsx:17]: Can they get two different roles?

File: packages/core/src/components/BaseInput/__tests__/BaseInput.jest.tsx [hunk 6]
```diff
@@ -0,0 +1,109 @@
...
+describe("BaseInput", () => {
+  it("should render correctly", () => {
+    const { getByLabelText } = renderBaseInput();
+    expect(getByLabelText("base-input")).toBeInTheDocument();
+  });
+
+  describe("with declared props", () => {
+    it("should apply the size class", () => {
+      const { getByLabelText } = renderBaseInput({ size: "large" });
+      expect(getByLabelText("base-input").parentNode).toHaveClass("large");
```
Comment (talkor) [inline packages/core/src/components/BaseInput/__tests__/BaseInput.jest.tsx:21]: I think you can use `container` or something similar to get the outer element

File: packages/core/src/components/BaseInput/__tests__/BaseInput.jest.tsx [hunk 7]
```diff
@@ -0,0 +1,109 @@
...
+      userEvent.type(input, "Hello, World!");
+      expect(input).toHaveValue("Hello, World!");
+    });
+
+    it("should call onChange on every input", () => {
+      const onChange = jest.fn();
+      const { getByLabelText } = renderBaseInput({ onChange });
+      const input = getByLabelText("base-input");
+      userEvent.type(input, "Hello, World!");
+      expect(onChange).toHaveBeenCalledTimes("Hello, World!".length);
```
Comment (talkor) [inline packages/core/src/components/BaseInput/__tests__/BaseInput.jest.tsx:82]: Extract to a variable?

File: packages/core/src/components/BaseInput/__tests__/BaseInput.jest.tsx [hunk 8]
```diff
@@ -0,0 +1,109 @@
...
+    });
+
+    it("should handle focus and blur events", () => {
```
Comment (talkor) [inline packages/core/src/components/BaseInput/__tests__/BaseInput.jest.tsx:85]: Nice!

**Review by talkor — APPROVED** [review]
Very cool!

File: packages/core/src/components/BaseInput/BaseInput.module.scss (hunk 4 above)
Comment (YossiSaadi) [reply packages/core/src/components/BaseInput/BaseInput.module.scss]: added support for both browsers that do and do not support has

File: packages/core/src/components/BaseInput/BaseInput.tsx (hunk 5 above)
Comment (YossiSaadi) [reply packages/core/src/components/BaseInput/BaseInput.tsx:16]: yep, for example in combobox, the wrapper should be "search" and the input "combobox" (it sounds weird, I know)

File: packages/core/src/components/BaseInput/__tests__/BaseInput.jest.tsx (hunk 6 above)
Comment (YossiSaadi) [reply packages/core/src/components/BaseInput/__tests__/BaseInput.jest.tsx:21]: container gives where the BaseInput is rendered, so I need to query the there the `firstChild` so it is pretty much the same..
I'll leave it the way it is I think

**Pull request merged**